tests/components/motionmount/test_config_flow.py

```diff
@@ -399,28 +373,216 @@ async def test_show_zeroconf_form_new_ce_new_pro(
         data=discovery_info,
     )
 
-    assert result["step_id"] == "zeroconf_confirm"
     assert result["type"] is FlowResultType.FORM
-    assert result["description_placeholders"] == {CONF_NAME: "My MotionMount"}
+    assert result["step_id"] == "auth"
 
 
-async def test_zeroconf_device_exists_abort(
+async def test_authentication_incorrect_pin(
     hass: HomeAssistant,
     mock_config_entry: MockConfigEntry,
     mock_motionmount_config_flow: MagicMock,
 ) -> None:
-    """Test we abort zeroconf flow if device already configured."""
+    """Test that authentication is requested when needed."""
+    type(mock_motionmount_config_flow).name = PropertyMock(return_value=ZEROCONF_NAME)
+    type(mock_motionmount_config_flow).mac = PropertyMock(return_value=MAC)
+    type(mock_motionmount_config_flow).is_authenticated = PropertyMock(
+        return_value=False
+    )
+
+    user_input = MOCK_USER_INPUT.copy()
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+        data=user_input,
+    )
+
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "auth"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input=MOCK_PIN_INPUT.copy(),
+    )
+
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "auth"
+
+    assert result["errors"]
+    assert result["errors"][CONF_PIN] == CONF_PIN
+
+
+async def test_authentication_first_incorrect_pin_to_backoff(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_motionmount_config_flow: MagicMock,
+) -> None:
+    """Test that authentication is requested when needed."""
+    type(mock_motionmount_config_flow).name = PropertyMock(return_value=ZEROCONF_NAME)
+    type(mock_motionmount_config_flow).mac = PropertyMock(return_value=MAC)
+    type(mock_motionmount_config_flow).is_authenticated = PropertyMock(
+        return_value=False
+    )
+    type(mock_motionmount_config_flow).can_authenticate = PropertyMock(
+        side_effect=[True, 4]
+    )
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+        data=MOCK_USER_INPUT.copy(),
+    )
+
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "auth"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input=MOCK_PIN_INPUT.copy(),
+    )
+
+    assert mock_motionmount_config_flow.authenticate.called
+    assert result["type"] is FlowResultType.SHOW_PROGRESS
+    assert result["step_id"] == "backoff"
```

Comment:
idem

Reply:
SHOW_PROGRESS is a kind of FORM, isn't it?